contracts/src/utils/Factory.sol

```diff
@@ -35,34 +39,35 @@ abstract contract Factory {
     bytes memory initCode,
     bytes32 salt
   ) internal returns (address deployment) {
-    assembly {
+    assembly ("memory-safe") {
       let encoded_data := add(0x20, initCode)
       let encoded_size := mload(initCode)
       deployment := create2(0, encoded_data, encoded_size, salt)
+      if iszero(deployment) {
+        mstore(0, 0xef35ca19) // revert Factory__FailedDeployment()
+        revert(0x1c, 0x04)
+      }
     }
-
-    if (deployment == address(0)) revert Factory__FailedDeployment();
   }
 
   /**
    * @notice calculate the _deployMetamorphicContract deployment address for a given salt
    * @param initCodeHash hash of contract initialization code
    * @param salt input for deterministic address calculation
-   * @return deployment address
+   * @return deployment deployment address
```

Comment:
double deployment?

Reply:
Is this to name the variable?

Reply:
Yes. This is to suppress compiler warning.